tests/components/wake_on_lan/test_switch.py

```diff
@@ -1,266 +1,277 @@
 """The tests for the wake on lan switch platform."""
-import unittest
-
-import homeassistant.components.switch as switch
-from homeassistant.const import STATE_OFF, STATE_ON
-from homeassistant.setup import setup_component
-
-from tests.async_mock import Mock, patch
-from tests.common import get_test_home_assistant, mock_service
-from tests.components.switch import common
-
-TEST_STATE = None
-
-
-def call(cmd, stdout, stderr):
-    """Return fake subprocess return codes."""
-    if cmd[5] == "validhostname" and TEST_STATE:
-        return 0
-    return 2
+import platform
+import subprocess
 
+import pytest
 
-def system():
-    """Fake system call to test the windows platform."""
-    return "Windows"
+import homeassistant.components.switch as switch
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.setup import async_setup_component
+
+from tests.async_mock import patch
+from tests.common import async_mock_service
+
+
+@pytest.fixture(autouse=True)
+def mock_send_magic_packet():
+    """Mock magic packet."""
+    with patch("wakeonlan.send_magic_packet") as mock_send:
+        yield mock_send
+
+
+async def test_valid_hostname(hass):
+    """Test with valid hostname."""
+    assert await async_setup_component(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6319846)